File: knapsack.cpp
#include<iostream>

using namespace std;

int main(){
    int weights[] = {0, 8, 2, 6, 1};
    int profit[] = {0, 50, 150, 210, 30};

    int data[5][11];

    const int n = 4;
    const int maxCapacity = 10;

    for (int itemNum = 0; itemNum <= n; itemNum++){
        for (int capacity = 0; capacity <= maxCapacity; capacity++){
            if (itemNum == 0 || capacity == 0)
            {
                data[itemNum][capacity] = 0;
            }
            else if(weights[itemNum] <= capacity)
            {
                data[itemNum][capacity] = std::max(profit[itemNum] + data[itemNum - 1][capacity - weights[itemNum]],
                                                data[itemNum - 1][ capacity]);
            }
            else
            {
                data[itemNum][capacity] = data[itemNum - 1][ capacity];
            }
        }
    }
    std::cout<<"Max value of the items included: "<<data[n][ maxCapacity]<<endl;
}

File: knapsack.py
def knapsack(weights, values, capacity):
    n = len(values)

    dp = [[0 for _ in range(capacity + 1)] for _ in range(n + 1)]

    for i in range(1, n + 1):
        for w in range(1, capacity + 1):
            current_weight = weights[i - 1]
            current_value = values[i - 1]

            if current_weight <= w:
                dp[i][w] = max(dp[i - 1][w], current_value + dp[i - 1][w - current_weight])
            else:
                dp[i][w] = dp[i - 1][w]

    return dp[n][capacity]


item_values = [60, 100, 120]
item_weights = [10, 20, 30]
knapsack_capacity = 50

max_value = knapsack(item_weights, item_values, knapsack_capacity)
print(f"Maximum value in Knapsack = {max_value}")  # Output: 220
